tidy3d/components/medium.py

```diff
@@ -2927,6 +2927,51 @@ class CustomAnisotropicMediumInternal(CustomAnisotropicMedium):
     )
 
 
+class AbstractNonlinearMedium(AbstractMedium, ABC):
+    """Abstract nonlinear medium.
+
+    Note
```

Comment:
not sure this note will propagate to the subclasses, so might need to copy it over.